lib/http/endpoint.js

```diff
@@ -76,6 +76,24 @@ class Context {
   with(other) { return new Context(this, other); }
 }
 
+// pulls out the data we want from the node request object. we do this as of node
+// v16 both because it saves data weight and because in v16 they changed the
+// request.headers property to be calculated, so we have to copy it manual.
+const getRequestContext = (request) => ({
+  method: request.method,
+  url: request.url,
+  originalUrl: request.originalUrl,
+  headers: request.headers,
```

Comment:
I'm struggling to understand how this works... does `request` now have a getter which returns the entire headers array?

Reply:
that's right